sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -274,30 +338,17 @@ def _check_categories(self, X):
                 if negative_categories.any():
                     categories = categories[~negative_categories]
 
-                if hasattr(self, "feature_names_in_"):
-                    feature_name = f"'{self.feature_names_in_[f_idx]}'"
-                else:
-                    feature_name = f"at index {f_idx}"
-
-                if categories.size > self.max_bins:
-                    raise ValueError(
-                        f"Categorical feature {feature_name} is expected to "
-                        f"have a cardinality <= {self.max_bins} but actually "
-                        f"has a cardinality of {categories.size}."
-                    )
-
-                if (categories >= self.max_bins).any():
-                    raise ValueError(
-                        f"Categorical feature {feature_name} is expected to "
-                        f"be encoded with values < {self.max_bins} but the "
-                        "largest value for the encoded categories is "
-                        f"{categories.max()}."
```

Comment:
Maybe, we should introduce a new parameter like `group_infrequent_categories: bool`. In case someone does not want this behavior. If False, error when `n_categories > max_bins`.
I think just describing the effect in the doc is not enough, therefore the parameter. Let's discuss.